src/mono/wasm/build/WasmApp.targets

```diff
@@ -308,15 +308,17 @@
 
   <Target Name="_GetWasmGenerateAppBundleDependencies">
     <PropertyGroup>
-      <WasmIcuDataFileName Condition="'$(InvariantGlobalization)' != 'true'">icudt.dat</WasmIcuDataFileName>
-
+      <IsIcuFilePredefined Condition="'$(InvariantGlobalization)' != 'true' and '$(IcuFileName)' != ''">true</IsIcuFilePredefined>
       <_HasDotnetWasm Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.wasm'">true</_HasDotnetWasm>
       <_HasDotnetJsWorker Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.worker.js'">true</_HasDotnetJsWorker>
       <_HasDotnetJsSymbols Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.js.symbols'">true</_HasDotnetJsSymbols>
       <_HasDotnetJs Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.js'">true</_HasDotnetJs>
     </PropertyGroup>
 
     <ItemGroup>
+      <WasmIcuDataFileNames Condition="'$(InvariantGlobalization)' != 'true' and '$(IsIcuFilePredefined)' != 'true'" Include="icudt.dat;icudt_EFIGS.dat;icudt_CJK.dat;icudt_no_CJK.dat"/>
```

Comment:
1) You're firght, `icudt.dat;icudt_EFIGS.dat;icudt_CJK.dat;icudt_no_CJK.dat` is a list of all available files. Why do we need to include them? We can load only one ICU file to an app. In case the user did not specify any value in the .csproj for ICU file, we need to have all of them ready in AppBundle to check the application locale on app startup and load to the memory only the ICU file that is in line with the app locale, so:
- EFIGS for "en", "fr-FR", "it-IT", "de-DE", "es-ES",
- CJK for "zh", "ko", "ja",
- "no_CJK" for any other.
Here I see I am making a mistake listing `icudt.dat` along with them. There's no option to load full ICU when we are choosing based on the app locale, thanks.

2) What's the difference between them and `icudt.dat`: `icudt.dat` is huge while each of: `icudt_EFIGS.dat;icudt_CJK.dat;icudt_no_CJK.dat` is small. We are loading only one of them to the application memory.

3) "I would roughly suggest:"
Do you suggest that the user should be able to set both: `WasmIncludeFullIcuData` and `IcuFileName` (alias `WasmIcuDataFile` )? It looks like a duplication of purpose. If we can set `icudt.dat` by passing it with `IcuFileName` then why to use `WasmIncludeFullIcuData` at all?

For now, I am removing `icudt.dat` from the list - it's incorrect to load it with smaller ICU files. I am not convinced about the other suggestions yet.

Reply:
Edit:
The fact that Blazor is already using `WasmIncludeFullIcuData` convinced me to apply it in this PR. As you suggested, it will load `icudt.dat`.
Remaining issues from this comment: 
- why would we want to introduce another flag `IcuAvailableDataFiles`? What are the advantages?
- `User can add to @(WasmIcuDataFile) to add custom files`:  We cannot let the user set the list, we want them to set a variable because we want to accept only one value. We should no allow "adding", only "setting". Or maybe it's only a misunderstanding, not an issue.

Reply:
Ok, I had misunderstood the design here. Is the following correct?

- Instead of including the full `icudt.dat`, we have split that into multiple parts like `icudt_EFIGS.dat;icudt_CJK.dat;icudt_no_CJK.dat` (lets call these `@(IcuAvailableDataFiles)`.
- This would allow loading the relevant one based on the application locale
- Only one `icu*dat` file will ever be loaded in the app
- Developers can create their own .dat files too, containing some custom set of cultures

- The behavior we want now when `InvariantGlobalization != true` is:
	- (a) by default, include all the `@(IcuAvailableDataFiles)` in the app bundle
  		- and when running the app, the runtime will decide which of the files to load based on the app environment
  	- (b) Alternatively, the user can choose to include the *full* set - `icudt.dat`
		- in this case the runtime will load that full file irrespective of the app environment

	- (c) As a 3rd case, the user can specify a custom file to always be loaded by the runtime.
		- this is similar to (b) in the sense that a single file is there in the bundle, and the runtime unconditionally loads that

Does the above sound correct?

Reply:
Thank you, it's correct. I think we are missing only one option in the last bullet point:
d) Similar to custom case but the user does not create their own, custom file. They choose one of our pre-prepared files: `icudt.dat;icudt_EFIGS.dat;icudt_CJK.dat;icudt_no_CJK.dat`. The difference is that they do not need to provide the full path for it because we know where it is located and where to copy it from
- this is similar to (b) and (c) in the sense that a single file is there in the bundle, and the runtime unconditionally loads that.

Reply:
Do you think that the docs need edition so that they will be describing the design more clearly?

Reply:
Ok, so based on this better understanding I would suggest:


- `@(IcuAvailableDataFile)` - all the shards, but does not include `icudt.dat`
- `$(WasmIcuFileName)` - single filename, can be a full path too
	- set by the user only
	- instead of modifying this property for computing the file to bundle, use a private property (name starting with `_`)

- `$(WasmBundleFullIcuData):`
	- when true - includes only `icudt.dat`
		- warn if WasmIcuFileName was set also
	- when false (default)
		- if WasmIcuFileName is ''
			- includes all the default shards
		- if WasmIcuFileName !is ''
			- includes only the specified file, whether a custom one, or one of the shards

- if `!InvariantGlobalization`:
	- based on icu properties, add to `@(WasmNativeAsset)`